### PR TITLE
NOISSUE - Add Python SDK

### PR DESCRIPTION
**Pull request by fbugarski**
Signed-off-by: Filip Bugarski <filipbugarski@gmail.com>

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,2 +1,127 @@
-# sdk-py
```
Comment (drasko) [inline README.md:1]: I think some kind of title will be needed. You can use `## Python SDK`

**Review by drasko — CHANGES_REQUESTED** [review]
Please remove all binary `.pyc` files

File: lib/errors.py [hunk 2]
```diff
@@ -0,0 +1,141 @@
+def handle_error(error_dict, status_code):
+    if status_code in error_dict.keys():
+        return error_dict[status_code]
+    elif status_code in errors.keys():
+        return errors[status_code]
+    else:
+        return "Unknown error"
+
```
Comment (dusanb94) [inline lib/errors.py:8]: Remove an empty line.

File: lib/errors.py [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+errors = {
+    400: "Failed due to malformed JSON.",
+    401: "Missing or invalid access token provided.",
+    403: "Missing or invalid access token provided.",
+    404: "A non-existent entity request.",
+    409: "Entity already exist.",
+    415: "Missing or invalid content type.",
+    422: "Database can't process request.",
+    500: "Unexpected server-side error occurred.",
+}
```
Comment (dusanb94) [inline lib/errors.py:19]: Add an empty line.

File: lib/errors.py [hunk 4]
```diff
@@ -0,0 +1,141 @@
...
+    "get": {
+        400: "Failed due to malformed query parameters.",
+    },
+    "get_all": {
+        400: "Failed due to malformed query parameters.",
+    },
+    "update": {
+        404: "Failed due to non existing user.",
+    },
+    "update_password": {
```
Comment (dusanb94) [inline lib/errors.py:36]: Please double-check all the empty errors. Also, please check if there are multiple possible errors for different actions (e.g. `malformed` and `conflict`).

File: lib/errors.py (hunk 2 above)
Comment (fbugarski) [reply lib/errors.py:8]: There should be two blank lines.

File: tests/test_version.py [hunk 5]
```diff
@@ -0,0 +1,11 @@
+from lib import sdk
+
+import json
+
+s = sdk.SDK()
+
+
```
Comment (drasko) [inline tests/test_version.py:7]: Kill this line

File: tests/test_users.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+user = {"email": "test@email.com", "password": "12345678"}
+user1 = {"email": "test1@email.com", "password": "12345678"}
+user2 = {"email": "test2@email.com", "password": "12345678"}
+old_password = {"old_password": "12345678"}
+password = {"password": "dsa"}
+user_id = "123-456"
+token = "9a8b7c6d5e4f3g21"
+url = "http://localhost"
+
+
```
Comment (drasko) [inline tests/test_users.py:15]: Why skipping two lines?

File: lib/response.py [hunk 7]
```diff
@@ -0,0 +1,11 @@
+class Error():
+    def __init__(self, status, message):
+        self.status = status
+        self.message = message
+
+
```
Comment (drasko) [inline lib/response.py:6]: Kill these whitelines

File: tests/test_channels.py [hunk 8]
```diff
@@ -0,0 +1,110 @@
+from lib import sdk
+
+import json
+
```
Comment (drasko) [inline tests/test_channels.py:4]: no need to skip 2 lines

File: tests/test_channels.py [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+s = sdk.SDK()
+
+channel = {"channel_name": "channel"}
+channel_id = "123-456-789"
+channel_id1 = "123-223-333"
+thing_id = "332-665-998"
+token = "9a8b7c6d5e4f3g21"
+url = "http://localhost"
+params = None
+
```
Comment (drasko) [inline tests/test_channels.py:15]: No need to skip 2 lines, and this should be corrected for the whole file

File: tests/test_messages.py [hunk 10]
```diff
@@ -0,0 +1,22 @@
+from lib import sdk
+
+
```
Comment (drasko) [inline tests/test_messages.py:3]: Same here

File: tests/test_things.py [hunk 11]
```diff
@@ -0,0 +1,138 @@
+from lib import sdk
+
+import json
+
```
Comment (drasko) [inline tests/test_things.py:4]: And here

**Review by drasko — CHANGES_REQUESTED** [review]

File: tests/test_users.py (hunk 6 above)
Comment (fbugarski) [reply tests/test_users.py:15]: There should be two blank lines.

File: tests/test_channels.py (hunk 9 above)
Comment (fbugarski) [reply tests/test_channels.py:14]: There should be two blank lines.

File: setup.cfg [hunk 12]
```diff
@@ -0,0 +1,22 @@
...
+long_description_content_type = text/markdown
+url = https://github.com/mainflux/mainflux/tree/master/pkg/sdk/python
+project_urls =
+    Bug Tracker = https://github.com/mainflux/mainflux/tree/master/pkg/sdk/python
+classifiers =
+    Programming Language :: Python :: 3.8.5
+    Operating System :: OS Independent
+
+[options]
+package_dir = src
```
Comment (drasko) [inline setup.cfg:17]: Is it `src` or `lib`?

File: setup.cfg [hunk 13]
```diff
@@ -0,0 +1,22 @@
...
+packages = find:
+python_requires = >=3.6
+
+[options.packages.find]
+where = src
```
Comment (drasko) [inline setup.cfg:22]: Same here

Comment (drasko) [reply setup.cfg:22]: Did you check if this setup / installation works?

**Review by drasko — CHANGES_REQUESTED** [review]

**Comment by drasko** [comment]
OK - I think we resolved everything, but that @fbugarski did not test setup / install of the package.

What I suggest at this point is that we merge this first implementation, and then open an issue for testing of the setup and `pip` installation, as well as for creating a release and uploading the package to PyPy.

**Comment by drasko** [comment]
Merged! Congrats @fbugarski!

**Pull request merged**